sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3488,6 +3511,15 @@ object functions {
    * Applies a binary operator to an initial state and all elements in the array,
    * and reduces this to a single state. The final state is converted into the final result
    * by applying a finish function.
+   * {{{
+   *   df.select(aggregate(col("i"), lit(0), (acc, x) => acc + x, _ * 10))
+   * }}}
+   *
+   * @param expr the input array column
+   * @param zero the initial value
+   * @param merge (c, v) => c, the merge function to reduce all elements into a single state
```

Comment:
`c`, `v` is too vague. How about `(combined_value, input_value) => combined_value`